Bottom: de2e26cac91c94592d1a4fa9d83bcae8bf662a4b
Top:    514e07fe7c74bee7ca24f8d3049b1f7d22bd981d
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-01-24 02:01:12 +0000

Add configure checks for GtkGLEext, add to CFLAGS and LIBS

*** TODO: Need separate tests for libgl and libglut etc.. ***









---

diff --git a/configure.ac b/configure.ac
index 7c6aa38..278e166 100644
--- a/configure.ac
+++ b/configure.ac
@@ -261,6 +261,45 @@ $DBUS_PKG_ERRORS])]
 
 fi
 
+
+AC_MSG_CHECKING([for whether to use GL drawing])
+AC_ARG_ENABLE([gl],
+[  --enable-gl           Enable GL drawing (with GTK HID)],
+[],[enable_gl=no])
+
+AC_MSG_RESULT([$enable_gl])
+AM_CONDITIONAL(USE_GL, test x$enable_gl = xyes)
+
+if test "x$enable_gl" = "xyes"; then
+	case " $with_gui " in
+		*\ gtk\ *) ;;
+		* ) AC_MSG_ERROR([GL drawing enabled but only works with the GTK HID.
+Either do not use --enable-gl or enable the gtk HID.])
+	;;
+	esac
+
+# FIXME... what do we need to check for determining we have GL? Is it headers, API ?
+
+	AC_CHECK_HEADERS(GL/gl.h)
+	case $ac_cv_header_GL_gl_h in
+	  no )
+	    AC_MSG_ERROR([You don't seem to have the GL library headers installed.])
+	    ;;
+	  * ) ;;
+	esac
+
+	AC_CHECK_HEADERS(GL/glu.h)
+	case $ac_cv_header_GL_glu_h in
+	  no )
+	    AC_MSG_ERROR([You don't seem to have the GL glu library headers installed.])
+	    ;;
+	  * ) ;;
+	esac
+
+	AC_DEFINE([ENABLE_GL], 1,
+		[Define to 1 if GL support is to be compiled in])
+fi
+
 AC_MSG_CHECKING([for which printer to use])
 AC_ARG_WITH([printer],
 [  --with-printer= 	  Specify the printer: lpr [[default=lpr]]],
@@ -504,6 +543,17 @@ Please review the following errors:
 $GLIB_PKG_ERRORS])]
 	)
 	GLIB_VERSION=`$PKG_CONFIG glib-2.0 --modversion`
+
+	if test "x$enable_gl" = "xyes"; then
+		# Check for GtkGLExt
+		PKG_CHECK_MODULES(GTKGLEXT, gtkglext-1.0 >= 1.0.0, , [AC_MSG_ERROR([
+*** Required version of gtkglext is not installed - please install first ***
+Please review the following errors:
+$GTKGLEXT_PKG_ERRORS])]
+		)
+	GTKGLEXT_VER=`$PKG_CONFIG gtkglext-1.0 --modversion`
+	fi
+
 	;;
 
       nelma|png )
@@ -751,8 +801,8 @@ AM_CONDITIONAL(DEBUG_BUILD, test x$enable_debug = xyes)
 
 # ------------- Complete set of CFLAGS and LIBS -------------------
 
-CFLAGS="$CFLAGS $X_CFLAGS $DBUS_CFLAGS $GTK_CFLAGS"
-LIBS="$LIBS $XM_LIBS $DBUS_LIBS $X_LIBS $GTK_LIBS $DMALLOC_LIBS $GD_LIBS $INTLLIBS"
+CFLAGS="$CFLAGS $X_CFLAGS $DBUS_CFLAGS $GTK_CFLAGS $GTKGLEXT_CFLAGS"
+LIBS="$LIBS $XM_LIBS $DBUS_LIBS $X_LIBS $GTK_LIBS $GTKGLEXT_LIBS $DMALLOC_LIBS $GD_LIBS $INTLLIBS -lglut"
 
 
 # if we have gcc then add -Wall
diff --git a/doc/actions.texi b/doc/actions.texi
index f8c4261..0a22627 100644
--- a/doc/actions.texi
+++ b/doc/actions.texi
@@ -1,5 +1,5 @@
 @c key actions
-@c ./../src/action.c 211
+@c ./../src/action.c 212
 
 Many actions take a @code{delta} parameter as the last parameter,
 which is an amount to change something.  That @code{delta} may include
@@ -18,7 +18,7 @@ Action(Object,-1)
 Actions which take a @code{delta} parameter which do not accept all
 these options will specify what they do take.
 
-@c ./../src/action.c 215
+@c ./../src/action.c 216
 
 @macro pinshapes
 
@@ -46,7 +46,7 @@ is documented for that purpose.
 
 @end macro
 
-@c ./../src/action.c 213
+@c ./../src/action.c 214
 
 Many actions act on indicated objects on the board.  They will have
 parameters like @code{ToggleObject} or @code{SelectedVias} to indicate
@@ -128,6 +128,7 @@ Affects all objects which are both selected and of the @var{Type} specified.
 * m Action:: Loads a layout into the current buffer.
 * MarkCrosshair Action:: Set/Reset the Crosshair mark
 * Message Action:: Writes a message to the log window.
+* MinClearGap Action:: Ensures that polygons are a minimum distance from objects.
 * MinMaskGap Action:: Ensures the mask is a minimum distance from pins and pads.
 * Mode Action:: Change or use the tool mode.
 * MorphPolygon Action:: Converts dead polygon islands into separate polygons.
@@ -185,7 +186,7 @@ AddRats(AllRats|SelectedRats|Close)@end format
 @end cartouche
 
 Add one or more rat lines to the board.
-@c ./../src/action.c 3194
+@c ./../src/action.c 3203
 
 @table @code
 
@@ -229,7 +230,7 @@ Atomic(Save|Restore|Close|Block)@end format
 @end cartouche
 
 Save or restore the undo serial number.
-@c ./../src/action.c 1605
+@c ./../src/action.c 1607
 
 This action allows making multiple-action bindings into an atomic
 operation that will be undone by a single Undo command.  For example,
@@ -269,7 +270,7 @@ AutoPlaceSelected()@end format
 @end cartouche
 
 Auto-place selected components.
-@c ./../src/action.c 3341
+@c ./../src/action.c 3350
 
 Attempts to re-arrange the selected components such that the nets
 connecting them are minimized.  Note that you cannot undo this.
@@ -284,7 +285,7 @@ AutoRoute(AllRats|SelectedRats)@end format
 @end cartouche
 
 Auto-route some or all rat lines.
-@c ./../src/action.c 3364
+@c ./../src/action.c 3373
 
 @table @code
 
@@ -318,7 +319,7 @@ ChangeClearSize(Selected|SelectedObjects, delta)@end format
 @end cartouche
 
 Changes the clearance size of objects.
-@c ./../src/action.c 3582
+@c ./../src/action.c 3591
 
 If the solder mask is currently showing, this action changes the
 solder mask clearance.  If the mask is not showing, this action
@@ -335,7 +336,7 @@ ChangeDrillSize(SelectedPins|SelectedVias|Selected|SelectedObjects, delta)@end f
 @end cartouche
 
 Changes the drilling hole size of objects.
-@c ./../src/action.c 3524
+@c ./../src/action.c 3533
 
 
 @node ChangeFlag Action
@@ -352,7 +353,7 @@ flag = square | octagon | thermal | join
 @end cartouche
 
 Sets or clears flags on objects.
-@c ./../src/action.c 5561
+@c ./../src/action.c 5668
 
 Toggles the given flag on the indicated object(s).  The flag may be
 one of the flags listed above (square, octagon, thermal, join).  The
@@ -369,7 +370,7 @@ ChangeHole(ToggleObject|Object|SelectedVias|Selected)@end format
 @end cartouche
 
 Changes the hole flag of objects.
-@c ./../src/action.c 4385
+@c ./../src/action.c 4492
 
 The "hole flag" of a via determines whether the via is a
 plated-through hole (not set), or an unplated hole (set).
@@ -384,7 +385,7 @@ ChangeJoin(ToggleObject|SelectedLines|SelectedArcs|Selected)@end format
 @end cartouche
 
 Changes the join (clearance through polygons) of objects.
-@c ./../src/action.c 3999
+@c ./../src/action.c 4106
 
 The join flag determines whether a line or arc, drawn to intersect a
 polygon, electrically connects to the polygon or not.  When joined,
@@ -403,7 +404,7 @@ ChangeName(Object)
 @end cartouche
 
 Sets the name of objects.
-@c ./../src/action.c 3802
+@c ./../src/action.c 3909
 
 @table @code
 
@@ -429,7 +430,7 @@ ChangeOctagon(SelectedElements|SelectedPins|SelectedVias)@end format
 @end cartouche
 
 Changes the octagon-flag of pins and vias.
-@c ./../src/action.c 4211
+@c ./../src/action.c 4318
 
 @pinshapes
 
@@ -443,7 +444,7 @@ ChangePaste(ToggleObject|Object|SelectedPads|Selected)@end format
 @end cartouche
 
 Changes the no paste flag of objects.
-@c ./../src/action.c 4427
+@c ./../src/action.c 4534
 
 The "no paste flag" of a pad determines whether the solderpaste
  stencil will have an opening for the pad (no set) or if there wil be
@@ -460,7 +461,7 @@ ChangePinName(ElementName,PinNumber,PinName)@end format
 @end cartouche
 
 Sets the name of a specific pin on a specific element.
-@c ./../src/action.c 3726
+@c ./../src/action.c 3833
 
 This can be especially useful for annotating pin names from a
 schematic to the layout without requiring knowledge of the pcb file
@@ -484,7 +485,7 @@ ChangeSize(SelectedElements, delta)@end format
 @end cartouche
 
 Changes the size of objects.
-@c ./../src/action.c 3435
+@c ./../src/action.c 3444
 
 For lines and arcs, this changes the width.  For pins and vias, this
 changes the overall diameter of the copper annulus.  For pads, this
@@ -504,7 +505,7 @@ ChangeSquare(Selected|SelectedObjects)@end format
 @end cartouche
 
 Changes the square flag of pins and pads.
-@c ./../src/action.c 4052
+@c ./../src/action.c 4159
 
 Note that @code{Pins} means both pins and pads.
 
@@ -521,7 +522,7 @@ ClearOctagon(SelectedElements|SelectedPins|SelectedVias)@end format
 @end cartouche
 
 Clears the octagon-flag of pins and vias.
-@c ./../src/action.c 4327
+@c ./../src/action.c 4434
 
 @pinshapes
 
@@ -535,7 +536,7 @@ ClearSquare(ToggleObject|SelectedElements|SelectedPins)@end format
 @end cartouche
 
 Clears the square-flag of pins and pads.
-@c ./../src/action.c 4158
+@c ./../src/action.c 4265
 
 Note that @code{Pins} means pins and pads.
 
@@ -555,7 +556,7 @@ flag = square | octagon | thermal | join@end format
 @end cartouche
 
 Clears flags on objects.
-@c ./../src/action.c 5544
+@c ./../src/action.c 5651
 
 Turns the given flag off, regardless of its previous setting.  See
 @code{ChangeFlag}.
@@ -574,7 +575,7 @@ Connection(Find|ResetLinesAndPolygons|ResetPinsAndVias|Reset)@end format
 @end cartouche
 
 Searches connections of the object at the cursor position.
-@c ./../src/action.c 2025
+@c ./../src/action.c 2027
 
 Connections found with this action will be highlighted in the
 ``connected-color'' color and will have the ``found'' flag set.
@@ -610,7 +611,7 @@ Delete(AllRats|SelectedRats)  ;@end format
 @end cartouche
 
 Delete stuff.
-@c ./../src/action.c 3258
+@c ./../src/action.c 3267
 
 
 @node DeleteRats Action
@@ -622,7 +623,7 @@ DeleteRats(AllRats|Selected|SelectedRats)@end format
 @end cartouche
 
 Delete rat lines.
-@c ./../src/action.c 3307
+@c ./../src/action.c 3316
 
 
 @node DisableVendor Action
@@ -654,7 +655,7 @@ DisperseElements(All|Selected)@end format
 @end cartouche
 
 Disperses elements.
-@c ./../src/action.c 2071
+@c ./../src/action.c 2073
 
 Normally this is used when starting a board, by selecting all elements
 and then dispersing them.  This scatters the elements around the board
@@ -681,7 +682,7 @@ Display(Pinout|PinOrPadName)
 @end cartouche
 
 Several display-related actions.
-@c ./../src/action.c 2189
+@c ./../src/action.c 2191
 
 @table @code
 
@@ -867,7 +868,7 @@ DRC()@end format
 @end cartouche
 
 Invoke the DRC check.
-@c ./../src/action.c 1640
+@c ./../src/action.c 1642
 
 Note that the design rule check uses the current board rule settings,
 not the current style settings.
@@ -882,7 +883,7 @@ DumpLibrary()@end format
 @end cartouche
 
 Display the entire contents of the libraries.
-@c ./../src/action.c 1674
+@c ./../src/action.c 1676
 
 
 
@@ -917,7 +918,7 @@ ExecuteFile(filename)@end format
 @end cartouche
 
 Run actions from the given file.
-@c ./../src/action.c 5675
+@c ./../src/action.c 5782
 
 Lines starting with @code{#} are ignored.
 
@@ -931,7 +932,7 @@ Flip(Object|Selected|SelectedElements)@end format
 @end cartouche
 
 Flip an element to the opposite side of the board.
-@c ./../src/action.c 1724
+@c ./../src/action.c 1726
 
 Note that the location of the element will be symmetric about the
 cursor location; i.e. if the part you are pointing at will still be at
@@ -1037,7 +1038,7 @@ LoadFrom(Layout|LayoutToBuffer|ElementToBuffer|Netlist|Revert,filename)@end form
 @end cartouche
 
 Load layout data from a file.
-@c ./../src/action.c 4839
+@c ./../src/action.c 4946
 
 This action assumes you know what the filename is.  The various GUIs
 should have a similar @code{Load} action where the filename is
@@ -1117,7 +1118,7 @@ MarkCrosshair()
 @end cartouche
 
 Set/Reset the Crosshair mark
-@c ./../src/action.c 3397
+@c ./../src/action.c 3406
 
 The ``mark'' is a small X-shaped target on the display which is
 treated like a second origin (the normal origin is the upper let
@@ -1139,7 +1140,7 @@ Message(message)@end format
 @end cartouche
 
 Writes a message to the log window.
-@c ./../src/action.c 1771
+@c ./../src/action.c 1773
 
 This action displays a message to the log window.  This action is primarily
 provided for use by other programs which may interface with PCB.  If
@@ -1147,6 +1148,23 @@ multiple arguments are given, each one is sent to the log window
 followed by a newline.
 
 
+@node MinClearGap Action
+@subsection MinClearGap
+@c key MinClearGap in hid 
+@cartouche
+@format
+MinClearGap(delta)
+"``MinClearGap(Selected, delta)@end format
+@end cartouche
+
+Ensures that polygons are a minimum distance from objects.
+@c ./../src/action.c 3735
+
+Checks all specified objects, and increases the polygon clearance if
+needed to ensure a minimum distance between their edges and the
+polygon edges.
+
+
 @node MinMaskGap Action
 @subsection MinMaskGap
 @c key MinMaskGap in hid 
@@ -1157,7 +1175,7 @@ MinMaskGap(delta)
 @end cartouche
 
 Ensures the mask is a minimum distance from pins and pads.
-@c ./../src/action.c 3650
+@c ./../src/action.c 3659
 
 Checks all specified pins and/or pads, and increases the mask if
 needed to ensure a minimum distance between the pin or pad edge and
@@ -1176,7 +1194,7 @@ Mode(Notify|Release|Cancel|Stroke)
 @end cartouche
 
 Change or use the tool mode.
-@c ./../src/action.c 2513
+@c ./../src/action.c 2522
 
 @table @code
 
@@ -1236,7 +1254,7 @@ MorphPolygon(Object|Selected)@end format
 @end cartouche
 
 Converts dead polygon islands into separate polygons.
-@c ./../src/action.c 3884
+@c ./../src/action.c 3991
 
 If a polygon is divided into unconnected "islands", you can use
 this command to convert the otherwise disappeared islands into
@@ -1304,7 +1322,7 @@ MoveObject(X,Y,dim)@end format
 @end cartouche
 
 Moves the object under the crosshair.
-@c ./../src/action.c 5376
+@c ./../src/action.c 5483
 
 The @code{X} and @code{Y} are treated like @code{delta} is for many
 other objects.  For each, if it's prefixed by @code{+} or @code{-},
@@ -1322,7 +1340,7 @@ MoveToCurrentLayer(Object|SelectedObjects)@end format
 @end cartouche
 
 Moves objects to the current layer.
-@c ./../src/action.c 5418
+@c ./../src/action.c 5525
 
 Note that moving an element from a component layer to a solder layer,
 or from solder to component, won't automatically flip it.  Use the
@@ -1378,7 +1396,7 @@ New([name])@end format
 @end cartouche
 
 Starts a new layout.
-@c ./../src/action.c 4902
+@c ./../src/action.c 5009
 
 If a name is not given, one is prompted for.
 
@@ -1415,7 +1433,7 @@ PasteBuffer(ToLayout, X, Y, units)@end format
 @end cartouche
 
 Various operations on the paste buffer.
-@c ./../src/action.c 4966
+@c ./../src/action.c 5073
 
 There are a number of paste buffers; the actual limit is a
 compile-time constant @code{MAX_BUFFER} in @file{globalconst.h}.  It
@@ -1474,7 +1492,7 @@ Polygon(Close|PreviousPoint)@end format
 @end cartouche
 
 Some polygon related stuff.
-@c ./../src/action.c 5312
+@c ./../src/action.c 5419
 
 Polygons need a special action routine to make life easier.
 
@@ -1558,7 +1576,7 @@ Quit()@end format
 @end cartouche
 
 Quits the application after confirming.
-@c ./../src/action.c 2003
+@c ./../src/action.c 2005
 
 If you have unsaved changes, you will be prompted to confirm (or
 save) before quitting.
@@ -1573,7 +1591,7 @@ Redo()@end format
 @end cartouche
 
 Redo recent``undo''operations.
-@c ./../src/action.c 5278
+@c ./../src/action.c 5385
 
 This routine allows you to recover from the last undo command.  You
 might want to do this if you thought that undo was going to revert
@@ -1596,7 +1614,7 @@ RemoveSelected()@end format
 @end cartouche
 
 Removes any selected objects.
-@c ./../src/action.c 2720
+@c ./../src/action.c 2729
 
 
 @node Renumber Action
@@ -1610,7 +1628,7 @@ Renumber()
 
 Renumber all elements.  The changes will be recorded to filename
 for use in backannotating these changes to the schematic.
-@c ./../src/action.c 2738
+@c ./../src/action.c 2747
 
 
 @node Report Action
@@ -1668,7 +1686,7 @@ RipUp(All|Selected|Element)@end format
 @end cartouche
 
 Ripup auto-routed tracks, or convert an element to parts.
-@c ./../src/action.c 3090
+@c ./../src/action.c 3099
 
 @table @code
 
@@ -1717,7 +1735,7 @@ RouteStyle(1|2|3|4)@end format
 @end cartouche
 
 Copies the indicated routing style into the current sizes.
-@c ./../src/action.c 5344
+@c ./../src/action.c 5451
 
 
 @node s Action
@@ -1749,7 +1767,7 @@ SaveSettings()
 @end cartouche
 
 Saves settings.
-@c ./../src/action.c 4823
+@c ./../src/action.c 4930
 
 If you pass no arguments, the settings are stored in
 @code{$HOME/.pcb/settings}.  If you pass the word @code{local} they're
@@ -1767,7 +1785,7 @@ SaveTo(PasteBuffer,filename)@end format
 @end cartouche
 
 Saves data to a file.
-@c ./../src/action.c 4733
+@c ./../src/action.c 4840
 
 @table @code
 
@@ -1807,7 +1825,7 @@ Select(TextByName|ViaByName, Name)
 @end cartouche
 
 Toggles or sets the selection
-@c ./../src/action.c 4469
+@c ./../src/action.c 4576
 
 @table @code
 
@@ -1856,7 +1874,7 @@ flag = square | octagon | thermal | join@end format
 @end cartouche
 
 Sets flags on objects.
-@c ./../src/action.c 5527
+@c ./../src/action.c 5634
 
 Turns the given flag on, regardless of its previous setting.  See
 @code{ChangeFlag}.
@@ -1875,7 +1893,7 @@ SetOctagon(Object|ToggleObject|SelectedElements|Selected)@end format
 @end cartouche
 
 Sets the octagon-flag of objects.
-@c ./../src/action.c 4269
+@c ./../src/action.c 4376
 
 @pinshapes
 
@@ -1889,7 +1907,7 @@ SetSame()@end format
 @end cartouche
 
 Sets current layer and sizes to match indicated item.
-@c ./../src/action.c 5459
+@c ./../src/action.c 5566
 
 When invoked over any line, arc, polygon, or via, this changes the
 current layer to be the layer that item is on, and changes the current
@@ -1905,7 +1923,7 @@ SetSquare(ToggleObject|SelectedElements|SelectedPins)@end format
 @end cartouche
 
 sets the square-flag of objects.
-@c ./../src/action.c 4105
+@c ./../src/action.c 4212
 
 Note that @code{Pins} means pins and pads.
 
@@ -1927,7 +1945,7 @@ Style = 2 has horizontal and vertical fingers with sharp edges.
 Style = 3 is a solid connection to the plane.Style = 4 has diagonal fingers with rounded edges.
 Style = 5 has horizontal and vertical fingers with rounded edges.
 
-@c ./../src/action.c 1797
+@c ./../src/action.c 1799
 
 This changes how/whether pins or vias connect to any rectangle or polygon
 on the current layer. The first argument can specify one object, or all
@@ -1953,7 +1971,7 @@ SetValue(Grid|Line|LineSize|Text|TextScale|ViaDrillingHole|Via|ViaSize, delta)@e
 @end cartouche
 
 Change various board-wide values and sizes.
-@c ./../src/action.c 1916
+@c ./../src/action.c 1918
 
 @table @code
 
@@ -1987,7 +2005,7 @@ ToggleHideName(Object|SelectedElements)@end format
 @end cartouche
 
 Toggles the visibility of element names.
-@c ./../src/action.c 3932
+@c ./../src/action.c 4039
 
 If names are hidden you won't see them on the screen and they will not
 appear on the silk layer when you print the layout.
@@ -2025,7 +2043,7 @@ Undo()
 @end cartouche
 
 Undo recent changes.
-@c ./../src/action.c 5117
+@c ./../src/action.c 5224
 
 The unlimited undo feature of @code{Pcb} allows you to recover from
 most operations that materially affect you work.  Calling
@@ -2069,7 +2087,7 @@ Unselect(TextByName|ViaByName)
 @end cartouche
 
 unselects the object at the pointer location or the specified objects
-@c ./../src/action.c 4619
+@c ./../src/action.c 4726
 
 @table @code
 
@@ -2256,7 +2274,7 @@ About()@end format
 @end cartouche
 
 Tell the user about this version of PCB.
-@c ./../src/hid/gtk/gtkhid-main.c 1615
+@c ./../src/hid/gtk/gtkhid-main.c 1614
 
 This just pops up a dialog telling the user which version of
 @code{pcb} they're running.
@@ -2287,7 +2305,7 @@ Center()
 @end cartouche
 
 Moves the pointer to the center of the window.
-@c ./../src/hid/gtk/gtkhid-main.c 2073
+@c ./../src/hid/gtk/gtkhid-main.c 2072
 
 Move the pointer to the center of the window, but only if it's
 currently within the window already.
@@ -2302,7 +2320,7 @@ Cursor(Type,DeltaUp,DeltaRight,Units)@end format
 @end cartouche
 
 Move the cursor.
-@c ./../src/hid/gtk/gtkhid-main.c 2172
+@c ./../src/hid/gtk/gtkhid-main.c 2171
 
 This action moves the mouse cursor.  Unlike other actions which take
 coordinates, this action's coordinates are always relative to the
@@ -2346,7 +2364,7 @@ DoWindows(Layout|Library|Log|Netlist|Preferences)@end format
 @end cartouche
 
 Open various GUI windows.
-@c ./../src/hid/gtk/gtkhid-main.c 2230
+@c ./../src/hid/gtk/gtkhid-main.c 2229
 
 @table @code
 
@@ -2400,7 +2418,7 @@ GetXY()@end format
 @end cartouche
 
 Get a coordinate.
-@c ./../src/hid/gtk/gtkhid-main.c 1630
+@c ./../src/hid/gtk/gtkhid-main.c 1629
 
 Prompts the user for a coordinate, if one is not already selected.
 
@@ -2416,7 +2434,7 @@ Popup(MenuName, [Button])@end format
 Bring up the popup menu specified by @code{MenuName}.
 If called by a mouse event then the mouse button number
 must be specified as the optional second argument.
-@c ./../src/hid/gtk/gtkhid-main.c 2297
+@c ./../src/hid/gtk/gtkhid-main.c 2296
 
 This just pops up the specified menu.  The menu must have been defined
 as a named subresource of the Popups resource in the menu resource
@@ -2433,7 +2451,7 @@ Print()@end format
 @end cartouche
 
 Print the layout.
-@c ./../src/hid/gtk/gtkhid-main.c 1948
+@c ./../src/hid/gtk/gtkhid-main.c 1947
 
 This will find the default printing HID, prompt the user for its
 options, and print the layout.
@@ -2448,7 +2466,7 @@ PrintCalibrate()@end format
 @end cartouche
 
 Calibrate the printer.
-@c ./../src/hid/gtk/gtkhid-main.c 1999
+@c ./../src/hid/gtk/gtkhid-main.c 1998
 
 This will print a calibration page, which you would measure and type
 the measurements in, so that future printouts will be more precise.
@@ -2466,7 +2484,7 @@ Save(PasteBuffer)@end format
 @end cartouche
 
 Save layout and/or element data to a user-selected file.
-@c ./../src/hid/gtk/gtkhid-main.c 1774
+@c ./../src/hid/gtk/gtkhid-main.c 1773
 
 This action is a GUI front-end to the core's @code{SaveTo} action
 (@pxref{SaveTo Action}).  If you happen to pass a filename, like
@@ -2499,7 +2517,7 @@ SetUnits(mm|mil)@end format
 @end cartouche
 
 Set the default measurement units.
-@c ./../src/hid/gtk/gtkhid-main.c 2269
+@c ./../src/hid/gtk/gtkhid-main.c 2268
 
 @table @code
 
@@ -2521,7 +2539,7 @@ SwapSides(|v|h|r)@end format
 @end cartouche
 
 Swaps the side of the board you're looking at.
-@c ./../src/hid/gtk/gtkhid-main.c 1856
+@c ./../src/hid/gtk/gtkhid-main.c 1855
 
 This action changes the way you view the board.
 
@@ -2663,7 +2681,7 @@ About()@end format
 @end cartouche
 
 Tell the user about this version of PCB.
-@c ./../src/hid/lesstif/dialogs.c 851
+@c ./../src/hid/lesstif/dialogs.c 870
 
 This just pops up a dialog telling the user which version of
 @code{pcb} they're running.
@@ -2678,7 +2696,7 @@ AdjustSizes()@end format
 @end cartouche
 
 Let the user change the board size, DRC parameters, etc
-@c ./../src/hid/lesstif/dialogs.c 1135
+@c ./../src/hid/lesstif/dialogs.c 1154
 
 Displays a dialog box that lets the user change the board
 size, DRC parameters, and text scale.
@@ -2842,7 +2860,7 @@ DoWindows(Layout|Library|Log|Netlist)@end format
 @end cartouche
 
 Open various GUI windows.
-@c ./../src/hid/lesstif/dialogs.c 815
+@c ./../src/hid/lesstif/dialogs.c 834
 
 @table @code
 
@@ -2895,7 +2913,7 @@ EditLayerGroups()@end format
 @end cartouche
 
 Let the user change the layer groupings
-@c ./../src/hid/lesstif/dialogs.c 1447
+@c ./../src/hid/lesstif/dialogs.c 1466
 
 Displays a dialog that lets the user view and change the layer
 groupings.  Each layer (row) can be a member of any one layer group
@@ -2915,7 +2933,7 @@ Export()@end format
 @end cartouche
 
 Export the layout.
-@c ./../src/hid/lesstif/dialogs.c 941
+@c ./../src/hid/lesstif/dialogs.c 960
 
 Prompts the user for an exporter to use.  Then, prompts the user for
 that exporter's options, and exports the layout.
@@ -3003,7 +3021,7 @@ Print()@end format
 @end cartouche
 
 Print the layout.
-@c ./../src/hid/lesstif/dialogs.c 878
+@c ./../src/hid/lesstif/dialogs.c 897
 
 This will find the default printing HID, prompt the user for its
 options, and print the layout.
@@ -3018,7 +3036,7 @@ PrintCalibrate()@end format
 @end cartouche
 
 Calibrate the printer.
-@c ./../src/hid/lesstif/dialogs.c 921
+@c ./../src/hid/lesstif/dialogs.c 940
 
 This will print a calibration page, which you would measure and type
 the measurements in, so that future printouts will be more precise.
diff --git a/doc/pcbfile.texi b/doc/pcbfile.texi
index 7dd6be8..70dc33a 100644
--- a/doc/pcbfile.texi
+++ b/doc/pcbfile.texi
@@ -926,4 +926,12 @@ Move items orthogonally.
 Draw autoroute paths real-time.
 @item 0x20000
 New polygons are full ones.
+@item 0x40000
+Names are locked, the mouse cannot select them.
+@item 0x80000
+Everything but names are locked, the mouse cannot select anything else.
+@item 0x100000
+New polygons are full polygons.
+@item 0x200000
+When set, element names are not drawn.
 @end table
diff --git a/doc/version.texi b/doc/version.texi
index bc949ab..ec116a2 100644
--- a/doc/version.texi
+++ b/doc/version.texi
@@ -1,4 +1,4 @@
-@set UPDATED 11 December 2007
-@set UPDATED-MONTH December 2007
+@set UPDATED 28 January 2009
+@set UPDATED-MONTH January 2009
 @set EDITION 1.99y
 @set VERSION 1.99y
